### PR TITLE
fix: update readme and change logic for checking home directory & user config file

### PR DESCRIPTION
**Pull request by brutusyhy**
This PR contains 3 changes:

1. Update the readme with a section for Windows users on how to install GNU make, in order to use the makefile scripts
2. Use homedir rather than "HOME" env variable to check the home directory, making it more platform independent
3. The app will not panic when it cannot find the user config file: it will consider the config to be empty.

### CODE REVIEW

File: internal/platform/configuration/src/user_settings.rs [hunk 1]
```diff
@@ -29,9 +29,15 @@ impl<'a> UserSettings {
     }
 
     pub fn load_configuration(&self, ctx: &mut Context) -> Result<ConfigurationModel> {
-        let mut file = ctx.block_on_with(self.fs_service.read_file(&self.resource))?;
+        // When the file doesn't exist, the code should not panic
```
Comment (jevgenis33) [inline internal/platform/configuration/src/user_settings.rs:32]: I think this comment is too obvious. Could you please delete it?

File: view/desktop/bin/src/commands/cmd_dummy.rs [hunk 2]
```diff
@@ -19,7 +20,14 @@ pub async fn fetch_all_themes() -> Result<Vec<String>, String> {
...
 #[specta::specta]
 pub async fn read_theme(theme_name: String) -> Result<String, String> {
-    let theme_file_path = PathBuf::from(std::env::var("HOME").unwrap())
+    let home_dir = if cfg!(target_family = "unix") {
+        PathBuf::from(std::env::var("HOME").unwrap())
+    } else {
+        my_home()
+            .unwrap()
+            .expect("Failed to retrieve the home directory")
+    };
```
Comment (g10z3r) [inline view/desktop/bin/src/commands/cmd_dummy.rs:29]: I think, here is better to use 

```rust
fn get_home_dir() -> Result<PathBuf, String> {
    #[cfg(target_os = "windows")]
    {
        windows_home_dir()
    }

    #[cfg(any(target_os = "macos", target_os = "linux"))]
    {
        unix_home_dir()
    }

    #[cfg(not(any(target_os = "windows", target_os = "macos", target_os = "linux")))]
    {
        Err("Unsupported operating system".to_string())
    }
}

/// Retrieves the home directory on Unix-like systems (macOS and Linux).
#[cfg(any(target_os = "macos", target_os = "linux"))]
fn unix_home_dir() -> Result<PathBuf, String> {
    std::env::var("HOME")
        .map(PathBuf::from)
        .map_err(|e| format!("Failed to retrieve HOME environment variable: {}", e))
}


/// Retrieves the home directory on Windows.
#[cfg(target_os = "windows")]
fn windows_home_dir() -> Result<PathBuf, String> {
    ...
}
```

Use example

```rust
 match get_home_dir() {
        Ok(home_dir) => todo!(),
        Err(err) =>  todo!(),,
    }
```

Comment (brutusyhy) [reply view/desktop/bin/src/commands/cmd_dummy.rs:29]: Great idea! I was thinking about making it into a function, but I'm not sure where I should put it. What's your thought?

File: internal/platform/configuration/src/user_settings.rs (hunk 1 above)
Comment (brutusyhy) [reply internal/platform/configuration/src/user_settings.rs]: Yes! I have deleted it!

**Review by g10z3r — APPROVED** [review]

**Pull request merged**